mk/compile.mk

```diff
@@ -230,4 +230,43 @@ endef
 
 $(foreach f,$(asm-defines-files),$(eval $(call gen-asm-defines-file,$(f),$(out-dir)/$(sm)/include/generated/$(basename $(notdir $(f))).h)))
 
+# Device tree source file compilation
+DTC 	:= dtc
+DTC_FLAGS += -I dts -O dtb
+DTC_FLAGS += -Wno-unit_address_vs_reg
+
+define gen-dtb-file
+# dts file path/name in $1
+# dtb file path/name in $2
+
+ifneq ($$(filter %.dts,$1),$1)
+$$(error gen-dtb-file expects a DTS file as first argument [$1])
+endif
+ifneq ($$(firstword $1),$(strip $1))
+$$(error gen-dtb-file expects a single entry as first argument [$1])
+endif
+
+dtb-predts-$2 	:= $$(dir $2)$$(notdir $2).pre.dts
+dtb-predep-$2 	:= $$(dir $2)$$(notdir $2).o.d
+dtb-dep-$2	:= $$(dir $2)$$(notdir $2).d
+
+dtb-cppflags-$2	:= -Icore/include/ -x assembler-with-cpp \
+		   -E -ffreestanding $$(CPPFLAGS) \
+		   -MD -MF $$(dtb-predep-$2) -MT $2
+
+dtb-dtcflags-$2	:= $$(DTC_FLAGS) -d $$(dtb-dep-$2)
+
+-include $$(dtb-dep-$2)
+-include $$(dtb-predep-$2)
+
+$2: $1
```

Comment:
Forgot one thing: why not do the `.cmd` file like above? I know its quite cumbersome, but it has a purpose, and compiling DTS files should not be different than compiling C files in that respect?

Reply:
I'm referring to the code that detects if the build command has changed: https://github.com/OP-TEE/optee_os/blob/master/mk/compile.mk#L128-L129

Reply:
Ok...